bmc/screenshot.go

```diff
@@ -0,0 +1,76 @@
+package bmc
+
+import (
+	"context"
+	"fmt"
+
+	bmclibErrs "github.com/bmc-toolbox/bmclib/v2/errors"
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+)
+
+// ScreenshotGetter interface provides methods to query for a BMC screen capture.
+type ScreenshotGetter interface {
+	Screenshot(ctx context.Context) (image []byte, fileType string, err error)
+}
+
+type screenshotGetterProvider struct {
+	name string
+	ScreenshotGetter
+}
+
+// screenshot returns an image capture of the video output.
+func screenshot(ctx context.Context, generic []screenshotGetterProvider) (image []byte, fileType string, metadata Metadata, err error) {
+	var metadataLocal Metadata
+
+	for _, elem := range generic {
+		if elem.ScreenshotGetter == nil {
+			continue
+		}
+		select {
+		case <-ctx.Done():
+			err = multierror.Append(err, ctx.Err())
+
+			return image, fileType, metadata, err
+		default:
+			metadataLocal.ProvidersAttempted = append(metadataLocal.ProvidersAttempted, elem.name)
+			image, fileType, vErr := elem.Screenshot(ctx)
+			if vErr != nil {
+				err = multierror.Append(err, errors.WithMessagef(vErr, "provider: %v", elem.name))
+				continue
+
+			}
+			metadataLocal.SuccessfulProvider = elem.name
+			return image, fileType, metadataLocal, nil
+		}
+	}
+
+	return image, fileType, metadataLocal, multierror.Append(err, errors.New("failed to capture screenshot"))
+}
+
+// ScreenshotFromInterfaces identifies implementations of the ScreenshotGetter interface and passes the found implementations to the screenshot() wrapper method.
+func ScreenshotFromInterfaces(ctx context.Context, generic []interface{}) (image []byte, fileType string, metadata Metadata, err error) {
```

Comment:
What needs this signature?

Reply:
Its how the client passes in the provider registry here - https://github.com/bmc-toolbox/bmclib/blob/49f837d37ccca73f85ca223d214988f36d2417f5/client.go#L364